Dockerfile

```diff
@@ -1,49 +1,25 @@
-FROM nvidia/cuda:11.3.1-base-ubuntu20.04
+FROM mambaorg/micromamba:focal-cuda-11.3.1
+# focal is Ubuntu 20.04
 
-# set the IGN proxy, otherwise apt-get and other applications don't work
-# Should be commented out outside of IGN
-ENV http_proxy 'http://192.168.4.9:3128/'
-ENV https_proxy 'http://192.168.4.9:3128/'
-
-# Remove any third-party apt sources to avoid issues with expiring keys.
-RUN rm -f /etc/apt/sources.list.d/*.list
-
-# Install some basic utilities
-RUN apt-get update && apt-get install -y \
-        nano \
-        curl \
-        ca-certificates \
-        sudo \
-        git \
-        bzip2 \
-        libx11-6 \
-        && rm -rf /var/lib/apt/lists/*
-
-# Create a working directory
-RUN mkdir /app
+WORKDIR /app
+COPY environment.yml environment.yml
 
-# Set up the Conda environment and make python accessible via PATH.
-ENV CONDA_AUTO_UPDATE_CONDA=false
-ENV PATH=/miniconda:/miniconda/bin:$PATH 
-COPY environment.yml /app/environment.yml
-RUN curl -sLo /miniconda.sh https://repo.continuum.io/miniconda/Miniconda3-py39_4.10.3-Linux-x86_64.sh \
-        && chmod +x /miniconda.sh \
-        && /miniconda.sh -b -p /miniconda \
-        && rm /miniconda.sh \
-        && /miniconda/bin/conda env update -n base -f /app/environment.yml \
-        && rm /app/environment.yml \
-        && /miniconda/bin/conda clean -ya
+RUN micromamba env create -f /app/environment.yml
 
+ENV PATH=$PATH:/opt/conda/envs/myria3d/bin/
 # Need to export this for torch_geometric to find where cuda is.
 # See https://github.com/pyg-team/pytorch_geometric/issues/2040#issuecomment-766610625
-ENV LD_LIBRARY_PATH="/miniconda/lib/:$LD_LIBRARY_PATH"
+ENV LD_LIBRARY_PATH="/opt/conda/envs/myria3d/lib/:$LD_LIBRARY_PATH"
 
-# Check succes of environment creation.
+# Check success of environment creation.
 RUN python -c "import torch_geometric;"
 
-# Copy the repository content in /app 
-WORKDIR /app
-COPY . .
+# use chown to prevent permission issues
+COPY --chown=$MAMBA_USER:MAMBA_USER . .
+# locate proj
+ENV PROJ_LIB=/opt/conda/envs/myria3d/share/proj/
+# Check taht myria3d can run
+RUN python -m myria3d.predict -h
```

Comment:
Here I would prefer keeping the `python run.py task.task_name=predict --help` syntax since this is the one we mostly use (cf. cicd.yaml workflow).

Reply:
updated